compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmCastGen.java

```diff
@@ -730,6 +734,17 @@ private static void generateCheckCastJToBJSON(MethodVisitor mv, BIRVarToJVMIndex
         generateJCastToBAny(mv, indexMap, sourceType, symbolTable.jsonType);
     }
 
+    private static void generateCheckCastJToBReadOnly(MethodVisitor mv, BIRVarToJVMIndexMap indexMap,
+                                                      JType sourceType) {
+
+        if (sourceType.jTag == JTypeTags.JREF || sourceType.jTag == JTypeTags.JARRAY) {
```

Comment:
can we say why these two types are skipped here ?

Reply:
I think the comment at L744 covers this already? Or should we still update this?

Btw the logic here is similar to other `generateCheckCastJToBXxxType` methods here.